likecoin/js/sidebar/src/pages/LikeCoinPlugin.js

```diff
@@ -0,0 +1,289 @@
+  }, [
+    props.DBLIKEPayAmount,
+    props.DBMemo,
+    props.DBArticleTitle,
+    props.DBAuthorDescription,
+    props.DBAuthor,
+    props.DBArticleURL,
+    props.DBwordCount,
+    props.DBArticleTags,
+    props.DBISCNId,
+    props.DBISCNVersion,
+    props.DBMattersArticleId,
+    props.DBMattersId,
+    props.DBMattersArticleSlug,
+  ]);
+  useEffect(() => {
+    setArweaveId(props.DBArweaveId);
+    setArweaveIPFSHash(props.DBArweaveIPFSHash);
+    setMattersIPFSHash(props.DBMattersIPFSHash);
+    const fingerprintsArr = [];
+    if (props.DBArweaveId && props.DBArweaveId !== '-') {
+      const arweaveFingerprint = `ar://${props.DBArweaveId}`;
+      fingerprintsArr.push(arweaveFingerprint);
+    }
+    if (props.DBArweaveIPFSHash && props.DBArweaveIPFSHash !== '-') {
+      const arweaveIPFSHashFingerprint = `ipfs://${props.DBArweaveIPFSHash}`;
+      fingerprintsArr.push(arweaveIPFSHashFingerprint);
+    }
+    if (props.DBMattersIPFSHash && props.DBMattersIPFSHash !== '-') {
+      const mattersIPFSHashFingerprint = `ipfs://${props.DBMattersIPFSHash}`;
+      fingerprintsArr.push(mattersIPFSHashFingerprint);
+    }
+    if (fingerprintsArr.length > 1) {
+      // at least have arweaveId & arweaveIPFS
+      setFingerPrints(fingerprintsArr);
+    }
+  }, [props.DBArweaveId, props.DBArweaveIPFSHash, props.DBMattersIPFSHash]);
+  useEffect(() => {
+    // Define functions that are used in useEffect()
+    async function onLIKEPaySuccessUploadToArweave(event) {
+      const { action, data } = JSON.parse(event.data);
+      if (action === 'TX_SUBMITTED') {
+        // Upload to Arweave
+        try {
+          await uploadToArweave(data); // Upload to Arweave & Generate full fingerprints.
+        } catch (error) {
+          console.error(error);
+        }
+      }
+    }
+    async function uploadToArweave(data) {
+      try {
+        const { tx_hash: txHash } = data;
+        props.postArweaveUploadAndIPFSData({ txHash });
+      } catch (error) {
+        console.error('Error occurs when uploading to Arweave:');
+        console.error(error);
+      }
+    }
+    async function onRegisterISCNSuccess(event) {
+      const { action, data } = JSON.parse(event.data);
+      const {
+        tx_hash: txHash, iscnId, iscnVersion, timestamp,
+      } = data;
+      if (action === 'ISCN_SUBMITTED') {
+        props.postISCNInfoData({
+          iscnHash: txHash,
+          iscnId,
+          iscnVersion,
+          timestamp,
+        });
+      }
+    }
+    // ---- Full flow. Article is not on Arweave net yet. Start LIKEPay
+    if (shouldStartProcess && parseFloat(LIKE) > 0) setShouldStartLIKEPay(true);
+    // ---- Direct register ISCN flow.
+    // If user has uploaded to Arweave before, then directly submit to ISCN
+    if (
+      shouldStartProcess
+      && parseFloat(LIKE) === 0
+      && arweaveIPFSHash
+      && arweaveIPFSHash.length > 1
+    ) {
+      setShouldDirectRegisterISCN(true);
```

Comment:
`shouldSkipArweave` would be better, also what is differecent between `setShouldDirectRegisterISCN(true)` and `setShouldIndirectRegisterISCN(false)`?

Reply:
`parseFloat(LIKE) === 0` never use `===` on float

Reply:
ok! Use BigNumber instead.